src/allencell_ml_segmenter/core/file_input_widget.py

```diff
@@ -214,19 +214,24 @@ def _from_directory_slot(self) -> None:
         self._model.set_input_mode(InputMode.FROM_PATH)
 
     def _update_layer_list(self, event: Optional[NapariEvent] = None) -> None:
+        existing_selection = [
+            self._viewer.get_layers_nonthreshold()[i].name
```

Comment:
How did the viewer become a member of the fileInputWidget?  

I see details about thresholding leaking into what is (i think) supposed to be a simple, generalized widget.

Reply:
FWIW i think ideally the interaction between fileInputWidgets and the Viewer happens via events.